pkg/resource/clickpipe.go

```diff
@@ -1520,43 +1849,235 @@ func (c *ClickPipeResource) syncClickPipeState(ctx context.Context, state *model
+				for j, key := range mapping.SortingKeys {
+					sortingKeysList[j] = types.StringValue(key)
+				}
+				tableMappingModel.SortingKeys, _ = types.ListValue(types.StringType, sortingKeysList)
+			} else {
+				tableMappingModel.SortingKeys = types.ListNull(types.StringType)
+			}
+
+			// For table_engine, preserve null from state if it was null (API may return default)
+			if stateMapping != nil && stateMapping.TableEngine.IsNull() {
+				tableMappingModel.TableEngine = types.StringNull()
+			} else if mapping.TableEngine != nil && *mapping.TableEngine != "" {
+				tableMappingModel.TableEngine = types.StringValue(*mapping.TableEngine)
+			} else if stateMapping != nil {
+				tableMappingModel.TableEngine = stateMapping.TableEngine
+			} else {
+				tableMappingModel.TableEngine = types.StringNull()
+			}
+
+			tableMappingList[i] = tableMappingModel.ObjectValue()
+		}
+
+		postgresModel := models.ClickPipePostgresSourceModel{
+			Host:          types.StringValue(clickPipe.Source.Postgres.Host),
+			Port:          types.Int64Value(int64(clickPipe.Source.Postgres.Port)),
+			Database:      types.StringValue(clickPipe.Source.Postgres.Database),
+			Settings:      settingsModel.ObjectValue(),
+			TableMappings: types.ListNull(models.ClickPipePostgresTableMappingModel{}.ObjectType()),
+		}
+
+		if len(tableMappingList) > 0 {
+			postgresModel.TableMappings, _ = types.ListValue(models.ClickPipePostgresTableMappingModel{}.ObjectType(), tableMappingList)
+		}
+
+		// Preserve credentials from state as API doesn't return them
+		if !statePostgresModel.Credentials.IsNull() {
+			postgresModel.Credentials = statePostgresModel.Credentials
+		} else {
+			postgresModel.Credentials = types.ObjectNull(models.ClickPipeSourceCredentialsModel{}.ObjectType().AttrTypes)
+		}
+
+		sourceModel.Postgres = postgresModel.ObjectValue()
+	} else {
+		sourceModel.Postgres = types.ObjectNull(models.ClickPipePostgresSourceModel{}.ObjectType().AttrTypes)
 	}
 
-	stateDestinationModel := models.ClickPipeDestinationModel{}
+	state.Source = sourceModel.ObjectValue()
+
+	// Check if this is a Postgres CDC pipe
+	isPostgresPipe := clickPipe.Source.Postgres != nil
 
+	stateDestinationModel := models.ClickPipeDestinationModel{}
 	if !state.Destination.IsNull() {
 		if diags := state.Destination.As(ctx, &stateDestinationModel, basetypes.ObjectAsOptions{}); diags.HasError() {
 			return fmt.Errorf("error reading ClickPipe destination: %v", diags)
 		}
 	}
 
+	destinationModel := models.ClickPipeDestinationModel{
+		Database: types.StringValue(clickPipe.Destination.Database),
+	}
+
+	// For Postgres CDC, the API doesn't return table/managedTable/columns, so preserve state values
+	if isPostgresPipe {
+		destinationModel.Table = stateDestinationModel.Table
+		destinationModel.ManagedTable = stateDestinationModel.ManagedTable
+		destinationModel.Columns = stateDestinationModel.Columns
+		destinationModel.TableDefinition = stateDestinationModel.TableDefinition
```

Comment:
Should those be nulled out as they're not applicable?

Reply:
To be more clear: for CDC pipes these are always null, so if state has a random value by mistake, it has to be overridden with null